test/module/shared_model/validators/protobuf/proto_tx_validator_test.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+#include "module/shared_model/validators/validators_fixture.hpp"
+#include "validators/protobuf/proto_transaction_validator.hpp"
+
+class ProtoTxValidatorTest : public ValidatorsTest {
+ protected:
+  iroha::protocol::Transaction generateEmptyTransaction() {
+    std::string creator_account_id = "admin@test";
+
+    TestTransactionBuilder builder;
+    auto tx = builder.creatorAccountId(creator_account_id)
+                  .createdTime(created_time)
+                  .quorum(1)
+                  .build()
+                  .getTransport();
+    return tx;
+  }
+
+  iroha::protocol::Transaction generateCreateRoleTransaction(
+      const std::string &role_name,
+      iroha::protocol::RolePermission permission) {
+    auto tx = generateEmptyTransaction();
+
+    iroha::protocol::CreateRole cr;
+    cr.set_role_name(role_name);
+    cr.add_permissions(permission);
+
+    tx.mutable_payload()
+        ->mutable_reduced_payload()
+        ->add_commands()
+        ->mutable_create_role()
+        ->CopyFrom(cr);
+    return tx;
+  }
+};
+
+/**
+ * @given iroha::protocol::Transaction with defined command
+ * @when it is validated
+ * @then answer with no errors is returned
+ */
+TEST_F(ProtoTxValidatorTest, CommandIsSet) {
+  auto tx = generateEmptyTransaction();
+  // add not set command
```

Comment:
Why `add no set`, when the test is called `CommandIsSet` ? Maybe use only TestTransactionBuilder in this test?